test-framework/maven/src/main/java/io/quarkus/maven/it/verifier/RunningInvoker.java

```diff
@@ -88,11 +88,12 @@ public void stop() {
         if (result == null) {
             return;
         }
+        // Kill all processes that were (indirectly) spawned by the current process.
+        // It's important to do it this way (instead of calling result.destroy() first)
+        // because otherwise children of that process can become orphaned zombies.
+        DevModeTestUtils.killDescendingProcesses();
```

Comment:
This invocation order problem took the most time to figure out. I was able to see (on Windows, via Process Explorer) that sometimes a sub-branch of processes survives `result.destroy()`, loosing that parent and thus also the transitive parent connection to the surefire test JVM process. That resulted in `descendants()` not finding those "orphaned" processes anymore.